library/core/src/iter/traits/iterator.rs

```diff
@@ -2535,29 +2699,41 @@ pub trait Iterator {
     /// ```
     #[inline]
     #[unstable(feature = "try_find", reason = "new API", issue = "63178")]
+    #[default_method_body_is_const]
     fn try_find<F, R>(&mut self, f: F) -> ChangeOutputType<R, Option<Self::Item>>
     where
         Self: Sized,
-        F: FnMut(&Self::Item) -> R,
-        R: Try<Output = bool>,
+        F: ~const FnMut(&Self::Item) -> R,
+        F: ~const Drop,
+        R: ~const Try<Output = bool>,
         R::Residual: Residual<Option<Self::Item>>,
-    {
-        #[inline]
-        fn check<I, V, R>(
-            mut f: impl FnMut(&I) -> V,
-        ) -> impl FnMut((), I) -> ControlFlow<R::TryType>
-        where
-            V: Try<Output = bool, Residual = R>,
-            R: Residual<Option<I>>,
-        {
-            move |(), x| match f(&x).branch() {
-                ControlFlow::Continue(false) => ControlFlow::CONTINUE,
-                ControlFlow::Continue(true) => ControlFlow::Break(Try::from_output(Some(x))),
-                ControlFlow::Break(r) => ControlFlow::Break(FromResidual::from_residual(r)),
-            }
+        <<R as Try>::Residual as Residual<Option<Self::Item>>>::TryType:
+            ~const Try<Output = Option<Self::Item>>,
+        <<R as Try>::Residual as Residual<Option<Self::Item>>>::TryType: ~const FromResidual,
+        Self::Item: ~const Drop,
+    {
+        pub struct Check<F>(F);
+
+        impl_const_closure! {
+            impl<F, I, V, R> const FnMut for Check<F>
+            where
+                F: ~const FnMut(&I) -> V,
+                F: ~const Drop,
+                I: ~const Drop,
+                V: ~const Try<Output = bool, Residual = R>,
+                R: Residual<Option<I>>,
+                R::TryType: ~const Try<Output = Option<I>>,
+                R::TryType: ~const FromResidual,
```

Comment:
Holy boilerplate. 😬

Maybe we need some "everything ~const" sugar?

Reply:
Adding `~const` super traits could possibly make this easier, however it needs work and we need to wait for it to land in bootstrap to incoporate its changes.